apps/frontend/src/app/3_organisms/TransactionStepDialog/TransactionStepDialog.types.ts

```diff
@@ -8,17 +10,77 @@ export interface TxConfig {
   gasLimit?: BigNumberish;
   gasPrice?: string;
   value?: BigNumberish;
+  // @deprecated use receipt.response instead
   hash?: string;
 }
 
-export interface Transaction {
+export type TxConf = {
+  transaction: Transaction;
+  receipt: TransactionReceipt;
+  config: TxConfig;
```

Comment:
```suggestion
  config: TransactionConfig;
```
Rename `TxConfig` as well.